corehq/apps/accounting/models.py

```diff
@@ -1111,6 +1111,7 @@ def save(self, *args, **kwargs):
         """
         Overloaded to update domain pillow with subscription information
         """
+        Subscription.get_active_subscription_by_domain.clear(Subscription, self.subscriber.domain)
```

Comment:
I don't think these are ever mutated, so we probably don't need to clear the cache when they are saved.  That said, it would be more future-proof to clear the cache.